File: examples/cpp-examples/MeshResolveDegenerations.dox.cpp
#include <MRMesh/MRMesh.h>
#include <MRMesh/MRMeshLoad.h>

#include <iostream>
#include <MRMesh/MRBox.h>
#include <MRMesh/MRMeshDecimate.h>

int main()
{
    auto mesh = MR::MeshLoad::fromAnySupportedFormat( "mesh.stl" );
    if ( !mesh )
    {
        std::cerr << mesh.error() << std::endl;
        return 1;
    }

    // you can set various parameters for the resolving process; see the documentation for more info
    auto changed = MR::resolveMeshDegenerations( *mesh, {
        .maxDeviation = 1e-5f * mesh->computeBoundingBox().diagonal(),
        .tinyEdgeLength = 1e-3f,
    } );

    return 0;
}

File: examples/python-examples/MeshResolveDegenerations.dox.py
import sys

import meshlib.mrmeshpy as mrmeshpy

# Load mesh
mesh = mrmeshpy.loadMesh("mesh.stl")

# you can set various parameters for the resolving process; see the documentation for more info
params = mrmeshpy.ResolveMeshDegenSettings()
params.maxDeviation = 1e-5 * mesh.computeBoundingBox().diagonal()
params.tinyEdgeLength = 1e-3

changed = mrmeshpy.resolveMeshDegenerations(mesh, params)
if not changed:
    print("No changes were made")
    sys.exit(0)

mrmeshpy.saveMesh(mesh, "fixed_mesh.stl")
